MOBILE_CLIENT/ANDROID/_HIGH/APK_WIFI_API_PII/description.md

```diff
@@ -1,7 +1,51 @@
-The application is using requesting the `ACCESS_WIFI_STATE` interface and calling APIs like `getConnectionInfo` to access sensitive information about the Wi-Fi access point, like BSSID, SSID, and RSSI, and about the device, like MAC address and IP address.
+Android APK WiFi API allows access to Personal Identifiable Information (PII) such as network names and access points usage history, potentially leading to private data inference.
 
-This API is known to be abused to access PII information like:
+Applications using the `ACCESS_WIFI_STATE` permission and calling APIs like `getConnectionInfo` can access sensitive information about WiFi access points (such as BSSID, SSID, and RSSI) and potentially infer:
+
+* MAC Address which is a unique device identifier
+* Geolocation data by using surrounding WiFi access points
+* User movement history and social links inference relying on repeating patterns in WiFi access point usage
+
+### Examples
+
+=== "Java"
+
+ ```java
+ 
+    import android.content.Context;
+    import android.net.wifi.WifiInfo;
+    import android.net.wifi.WifiManager;
+    
+    public class WifiInfoRetriever {
+    
+        private Context context;
+    
+        public WifiInfoRetriever(Context context) {
+            this.context = context;
+        }
+    
+        public void retrieveWifiInfo() {
+            WifiManager wifiManager = (WifiManager) context.getSystemService(Context.WIFI_SERVICE);
+            
+            if (wifiManager.isWifiEnabled()) {
+                WifiInfo wifiInfo = wifiManager.getConnectionInfo();
+                
+                String ssid = wifiInfo.getSSID();
+                String bssid = wifiInfo.getBSSID();
+                int rssi = wifiInfo.getRssi();
+                String macAddress = wifiInfo.getMacAddress();
+                
+                System.out.println("SSID: " + ssid);
+                System.out.println("BSSID: " + bssid);
+                System.out.println("RSSI: " + rssi);
+                System.out.println("MAC Address: " + macAddress);
```

Comment:
Should be nicer to show collection here sending it to a domain.

Reply:
what do you mean?

Reply:
Done